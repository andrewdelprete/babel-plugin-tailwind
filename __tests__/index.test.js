import { transform } from "babel-core";
import plugin from "../src";

describe("Check", () => {
  test("selectors", () => {
    const code = "tw('text-red')";

    const actual = transform(code, {
      plugins: [plugin]
    }).code;

    expect(actual).toMatchSnapshot();
  });

  test(":hover selectors", () => {
    const code = "tw('w-5/6 hover:text-red hover:border-purple')";

    const actual = transform(code, {
      plugins: [plugin]
    }).code;

    expect(actual).toMatchSnapshot();
  });

  test("screen size selectors", () => {
    const code = "tw('w-5/6 sm:text-purple md:text-red md:border-purple')";

    const actual = transform(code, {
      plugins: [plugin]
    }).code;

    expect(actual).toMatchSnapshot();
  });

  test("screen size :hover selectors", () => {
    const code = "tw('w-5/6 md:hover:text-red lg:hover:border-purple')";

    const actual = transform(code, {
      plugins: [plugin]
    }).code;

    expect(actual).toMatchSnapshot();
  });

  test("various types of selectors", () => {
    const code =
      "tw('w-5/6 rounded-t-full text-red hover:text-purple hover:border-red sm:hover:text-blue sm:border-red sm:hover:border-black md:hover:w-1/3 md:hover:w-1/3')";

    const actual = transform(code, {
      plugins: [plugin]
    }).code;

    expect(actual).toMatchSnapshot();
  });

  test("custom selectors", () => {
    const code = "tw('w-5/6', { ':hover': {'borderColor': 'blue'} })";

    const actual = transform(code, {
      plugins: [plugin]
    }).code;

    expect(actual).toMatchSnapshot();
  });

  test("array of selectors", () => {
    const code = "tw(['w-5/6', 'md:text-red'])";

    const actual = transform(code, {
      plugins: [plugin]
    }).code;

    expect(actual).toMatchSnapshot();
  });
});
