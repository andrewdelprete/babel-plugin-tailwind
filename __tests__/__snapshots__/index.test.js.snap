// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check :focus selectors 1`] = `
"({
  ':focus': {
    'color': '#e3342f'
  }
});"
`;

exports[`Check :hover selectors 1`] = `
"({
  'width': '83.33333%',
  ':hover': {
    'color': '#e3342f',
    'borderColor': '#9561e2'
  }
});"
`;

exports[`Check array of selectors 1`] = `
"({
  'width': '83.33333%',
  '@media (min-width: 768px)': {
    'color': '#e3342f'
  }
});"
`;

exports[`Check custom selectors 1`] = `
"({
  'width': '83.33333%',
  ':hover': {
    'borderColor': 'blue'
  }
});"
`;

exports[`Check screen size :focus selectors 1`] = `
"({
  'width': '83.33333%',
  '@media (min-width: 768px)': {
    ':focus': {
      'color': '#e3342f'
    }
  },
  '@media (min-width: 992px)': {
    ':focus': {
      'borderColor': '#9561e2'
    }
  }
});"
`;

exports[`Check screen size :hover selectors 1`] = `
"({
  'width': '83.33333%',
  '@media (min-width: 768px)': {
    ':hover': {
      'color': '#e3342f'
    }
  },
  '@media (min-width: 992px)': {
    ':hover': {
      'borderColor': '#9561e2'
    }
  }
});"
`;

exports[`Check screen size selectors 1`] = `
"({
  'width': '83.33333%',
  '@media (min-width: 576px)': {
    'color': '#9561e2'
  },
  '@media (min-width: 768px)': {
    'color': '#e3342f',
    'borderColor': '#9561e2'
  }
});"
`;

exports[`Check selectors 1`] = `
"({
  'color': '#e3342f'
});"
`;

exports[`Check various types of selectors 1`] = `
"({
  'width': '83.33333%',
  'borderTopLeftRadius': '9999px',
  'borderTopRightRadius': '9999px',
  'color': '#e3342f',
  ':hover': {
    'color': '#9561e2',
    'borderColor': '#e3342f'
  },
  ':focus': {
    'borderColor': '#e3342f'
  },
  '@media (min-width: 576px)': {
    'borderColor': '#e3342f',
    ':hover': {
      'color': '#3490dc',
      'borderColor': '#22292f'
    }
  },
  '@media (min-width: 768px)': {
    ':hover': {
      'width': '33.33333%'
    }
  }
});"
`;
